netqasm/sdk/builder.py

```diff
@@ -1500,7 +1622,8 @@ def _build_cmds_loop_body(
             step=step,
             loop_register=loop_register,
         )
-        self._mem_mgr.remove_active_register(loop_register)
+        if not loop_register_already_activated:
```

Comment:
The boolean check is not wrong but isn't it an idea to also include the check if reg in active_registers in remove_active_register

Reply:
Here we want to make sure that the active state (true or false) of loop_register is the same as at the start of this method. Therefore we check `if not loop_register_already_activated` twice. I don't think it's possible with only a check inside `remove_active_register`.